cmd/libs/go2idl/types/types.go

```diff
@@ -84,6 +84,14 @@ type Package struct {
 	// package name).
 	Types map[string]*Type
 
+	// Functions within this package, indexed by their name (*not* including
+	// package name).
+	Functions map[string]*Type
```

Comment:
Yeah, I think storing types and things-with-names (vars, non-anonymous funcs, consts) separately is the way to go.